samples/cpp/image_generation/text2image_concurrency.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+#include <iostream>
+
+#include "openvino/genai/image_generation/text2image_pipeline.hpp"
+
+
+#include "imwrite.hpp"
+#include "progress_bar.hpp"
+
+int32_t main(int32_t argc, char* argv[]) try {
+    OPENVINO_ASSERT(argc >= 3, "Usage: ", argv[0], " <MODEL_DIR> '<PROMPT>' '<PROMPT>' ...");
+
+    const std::string models_path = argv[1];
+    const std::string device = "CPU";  // GPU can be used as well
+
+    std::vector<std::string> prompts;
+    for (int i = 2; i < argc; ++i) {
+        prompts.push_back(argv[i]);
+    }
+
+    ov::genai::Text2ImagePipeline pipe(models_path, device);
+
+    std::vector<std::thread> threads;
+
+    for (size_t i = 0; i < prompts.size(); ++i) {
+        threads.emplace_back([i, &pipe, models_path, prompts] () {
```

Comment:
it's safer and clearer to pre-clone a pipeline (request) per thread and pass only the specific prompt and corresponding request to the thread:
```cpp
threads.emplace_back([i, prompt, &request]()
```

Reply:
Sure, I can change it, however please check this comment: https://github.com/openvinotoolkit/openvino.genai/pull/2190#discussion_r2102144487